app/policies/application_policy.rb

```diff
@@ -17,7 +17,7 @@ def show?
   end
 
   def create?
-    true
+    @user.is_a?(User)
```

Comment:
Here is the key change. Pretty much all other changes in this PR are downstream from this.